File: semester_4/web/lab_09/src/main/java/model/Authenticator.java
package model;

import java.sql.*;

public class Authenticator {
    private PreparedStatement preparedStatement;
    private Connection con;

    public Authenticator() {
        connect();
    }

    public void connect() {
        try {
            Class.forName("com.mysql.jdbc.Driver");
            con = DriverManager.getConnection("jdbc:mysql://localhost/tictactoe", "root", "");

        } catch(Exception e) {
            //System.out.println("Connection error:" + e.getMessage());
            e.printStackTrace();
        }
    }

    public boolean authenticate(String username, String password) {

        ResultSet rs;
        boolean result = false;
        System.out.println(username + " " + password);

        try {
            preparedStatement = con.prepareStatement("select * from users where username=?");
            preparedStatement.setString(1, username);
            rs = preparedStatement.executeQuery();

            if (rs.next() && rs.getString("password").equals(password)) {
                result = true;
            }

            rs.close();
        } catch (SQLException e) {
            e.printStackTrace();
        }

        return result;
    }
}

File: semester_4/web/New Folder/backend/Authenticator.cs
using System.Net;
using System.Net.Http;
using System.Web.Http;
using MySql.Data.MySqlClient;

namespace api
{
    public class Authenticator
    {
        private MySqlDatabase _database;

        public Authenticator(MySqlDatabase database)
        {
            _database = database;
        }

        public bool Authenticate(string username, string password)
        {
            MySqlCommand command = _database.Connection.CreateCommand() as MySqlCommand;
            command.CommandText = @"SELECT * FROM users WHERE username=@Username";
            command.Parameters.AddWithValue("@Username", username);
            MySqlDataReader reader = command.ExecuteReader();

            bool result = false;

            if (reader.Read())
            {
                if (password == reader.GetString("password"))
                {
                    result = true;
                }
            }

            reader.Close();

            return result;
        }
    }
}